tests/ci/run_s2n_integration.sh

```diff
@@ -1,25 +1,90 @@
-#!/bin/bash -ex
+#!/bin/bash -exu
 # Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-source tests/ci/common_posix_setup.sh
+# Set up environment.
 
-AWS_LC_DIR=${PWD##*/}
+AWS_LC_DIR_NAME=${PWD##*/}
 cd ../
 ROOT=$(pwd)
-mkdir -p aws-lc-build aws-lc-install s2n-tls-build
-git clone https://github.com/aws/s2n-tls.git
+AWS_LC_DIR=${ROOT}/${AWS_LC_DIR_NAME}
+
+SCRATCH_FOLDER=${ROOT}/"SCRATCH_AWSLC_S2N_INTERN_TEST"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+S2N_TLS_BUILD_FOLDER="${SCRATCH_FOLDER}/s2n-tls-build"
+
+# Make script execution idempotent.
+mkdir -p ${SCRATCH_FOLDER}
+rm -rf ${SCRATCH_FOLDER}/*
+cd ${SCRATCH_FOLDER}
+
+# Test helpers.
+
+function fail() {
+    echo "test failure: $1"
+    exit 1
+}
+
+function aws_lc_build() {
+	cmake ${AWS_LC_DIR} -GNinja "-B${AWS_LC_BUILD_FOLDER}" "-DCMAKE_INSTALL_PREFIX=${AWS_LC_INSTALL_FOLDER}" "$@"
+	ninja -C ${AWS_LC_BUILD_FOLDER} install
+	ls -R ${AWS_LC_INSTALL_FOLDER}
+	rm -rf ${AWS_LC_BUILD_FOLDER}/*
+}
+
+function s2n_tls_build() {
+	cmake s2n -GNinja "-B${S2N_TLS_BUILD_FOLDER}" "-DCMAKE_PREFIX_PATH=${AWS_LC_INSTALL_FOLDER}" "$@"
+	ninja -C ${S2N_TLS_BUILD_FOLDER}
+	ls -R ${S2N_TLS_BUILD_FOLDER}
+}
+
+function s2n_tls_run_tests() {
+	cd ${S2N_TLS_BUILD_FOLDER}
+	ctest -j 8 --output-on-failure
+	cd ${SCRATCH_FOLDER}
+}
+
+function s2n_tls_prepare_new_build() {
+	rm -rf ${S2N_TLS_BUILD_FOLDER}/*
+}
+
+# Get latest s2n-tls version.
+
+mkdir -p ${AWS_LC_BUILD_FOLDER} ${AWS_LC_INSTALL_FOLDER} ${S2N_TLS_BUILD_FOLDER}
+git clone https://github.com/torben-hansen/s2n.git
```

Comment:
Need to change this before merging. But not before https://github.com/aws/s2n-tls/pull/3346 is merged

Reply:
Fix merged https://github.com/aws/s2n-tls/commit/36932b2875c97495e6b41c9b1f1e9b43a6e4bf19

So, pointing this to the tip of `s2n-tls` main branch.